Fixes to SRLatch; add tests for SRLatch and Decoder.

adhameer opened the pull request.

adhameer committed:
Merge remote-tracking branch 'upstream/master' into develop

adhameer committed:
Fix a small typo in the README.

adhameer committed:
Add tests for Decoder.

adhameer committed:
Fix code repetition for Decoder.

adhameer committed:
Merge remote-tracking branch 'upstream/develop' into develop

adhameer committed:
General improvements to SRLatch.

curiousguy13 commented:
you might also want to improve the indentation in sequential.py according to the PEP8 convention.

salilkapur commented:
The typo that you tried to fix is actually not a typo. I would recommend you undo that.  setInput() is for the gate and not for SRLatch. SRLatch setInputs() works just fine.

Variable names look good now.

Can you also combine the different tests into one? We can have one test for Decoder that can cover all the possibilities.

adhameer committed:
Combined Decoder test cases into one test.

adhameer committed:
Updated sequential.py to conform to PEP8 guidelines.

adhameer committed:
Added a couple of Decoder tests; added sequential_tests.py.

adhameer committed:
Fixed some misplaced parentheses.

adhameer committed:
Added sequential_tests.py for testing.

adhameer commented:
I added the relevant Decoder tests into combinational_tests.py and updated sequential.py to conform to PEP8 guidelines.

I believe the setInputs() thing actually was a typo. Previously, when calling SRLatch.setInputs(), it was calling setInput() on each of the NOR gates it was made up of: for example, NOR.setInput(S, self.b). That was setting the input at index S of the NOR gate's inputs to self.b, which was resulting in pretty strange behaviour. I think the desired behaviour was to overwrite the inputs to the NOR gate with [S, self.b]. I changed it now so that it only updates the input at index 0, since the Connector at index 1 doesn't need to be changed.
